spec/freemail_spec.rb

```diff
@@ -22,6 +22,36 @@
       let(:email) { 'steve@apple.com' }
       it { is_expected.to be(false) }
     end
+
+    context 'with a user-name-only email' do
+      let(:email) { 'apple.com@' }
+      it { expect { subject }.to raise_error(ArgumentError, /Invalid email/) }
```

Comment:
It might be okay if it returns true or false value:

```ruby
# maybe true
it { is_expected.to be(true) }

# or maybe false
it { is_expected.to be(false) }
```

But I'm not sure which is desirable result.

Reply:
Good point! The answer may depend on whether it is paired with a separate validation in the consuming code.

I'm thinking that maybe it could act like `String#to_i` or another such type-cast/value-transform method, where an invalid string will still produce some output (e.g. `'asdf'.to_i => 0`) and it's on the developer to validate the inputs. In a Rails context, it's common for ActiveModel::Validations to do the heavy lifting, and email validation is complex enough that I think it would make sense for this library to avoid having an opinion about what constitutes a "valid email" and avoid raising exceptions.